encord/http/error_utils.py

```diff
@@ -15,92 +15,13 @@
     """
-    if response == AUTHENTICATION_ERROR:
-        raise AuthenticationError("Invalid API key.")
-
-    if response == AUTHORISATION_ERROR:
-        raise AuthorisationError("You are not authorised to access this asset.")
-
-    if response == RESOURCE_NOT_FOUND_ERROR:
-        if payload:
-            raise ResourceNotFoundError(payload)
-        raise ResourceNotFoundError("The requested resource was not found in the database.")
-
-    if response == METHOD_NOT_ALLOWED_ERROR:
-        raise MethodNotAllowedError("HTTP method is not allowed.")
-
-    if response == UNKNOWN_ERROR:
-        raise UnknownException("An unknown error occurred.")
-
-    if response == OPERATION_NOT_ALLOWED_ERROR:
-        raise OperationNotAllowed("The read/write operation is not allowed by the API key.")
-
-    if response == ANSWER_DICTIONARY_ERROR:
-        raise AnswerDictionaryError("An object or classification is missing in the answer dictionaries.")
-
-    if response == CORRUPTED_LABEL_ERROR:
-        raise CorruptedLabelError(
-            "The label blurb is corrupted. This could be due to the number of "
-            "frame labels exceeding the number of frames in the labelled video."
-        )
-
-    if response == FILE_TYPE_NOT_SUPPORTED_ERROR:
-        raise FileTypeNotSupportedError("Supported file types are: image/jpeg, image/png, video/webm, video/mp4.")
-
-    if response == FILE_SIZE_NOT_SUPPORTED_ERROR:
-        raise FileSizeNotSupportedError("The combined size of the input files is larger than the supported limit")
-
-    if response == FEATURE_DOES_NOT_EXIST_ERROR:
-        raise FeatureDoesNotExistError("The passed feature does not exist in the project ontology.")
-
-    if response == MODEL_WEIGHTS_INCONSISTENT_ERROR:
-        raise ModelWeightsInconsistentError("The passed model weights are incompatible with the selected model.")
-
-    if response == MODEL_FEATURES_INCONSISTENT_ERROR:
-        raise ModelFeaturesInconsistentError("The passed features are incompatible with the selected model.")
-
-    if response == UPLOAD_OPERATION_NOT_SUPPORTED_ERROR:
-        raise UploadOperationNotSupportedError(
-            "Uploading a file to an external (e.g. S3/GCP/Azure) dataset is not " "permitted."
-        )
-
-    if response == DETECTION_RANGE_INVALID_ERROR:
-        raise DetectionRangeInvalidError(
-            "The detection range is invalid (e.g. less than 0, or" " higher than num frames in the video)"
-        )
-
-    if response == INVALID_DATE_FORMAT_ERROR:
-        raise InvalidDateFormatError("Invalid date format supplied as input")
-
-    if response == RESOURCE_EXISTS_ERROR:
-        raise ResourceExistsError(
-            "Trying to create a resource that already exists. " "Payload for this failure is: " + str(payload)
-        )
-
-    pass
+    if payload is not None:
+        payload = " with the following message: " + payload
+    message = f"The server has reported a response of type `{response}`{payload}"
```

Comment:
I can't say I like the way this looks, stylistically. Modifying the parameter variable, really?

Also, if the payload *is* a `None`, then it will get printed as such, which is ugly-ish.